lib/Conversion/TritonGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -1099,8 +1244,21 @@ struct AtomicRMWOpConversion
 
     auto valueTy = op.getResult().getType();
     auto tensorTy = valueTy.dyn_cast<RankedTensorType>();
+    auto elementTy = tensorTy.getElementType();
+    const bool isHopper =
+        moduleOp->hasAttr("triton_gpu.compute-capability") &&
+        moduleOp->getAttrOfType<IntegerAttr>("triton_gpu.compute-capability")
+                .getInt() >= 90;
+    if (atomicRmwAttr == RMWOp::FADD && elementTy.isBF16() && !isHopper) {
+      if (valElements.size() && !valElements[0].getType().isBF16()) {
+        assert(false && "atom.add.bf16 fallback requires bf16 stored elements");
+      } else {
+        return matchAndRewriteROCm(op, adaptor, rewriter);
```

Comment:
Should this be called rewriteUsingCAS or something?

Reply:
Well so, based on my experimentation it only does an atomic cas because the nvptx backend seems to support only that pattern and not the more compact atom.global.add ptx (this goes for f16 and f32 as well). At the moment on ampere for bf16 there doesn't seem to be anything better I know of that we can do so the nvptx backends lowering is good enough as a fall back I think.